src/components/BulkEnrollmentResultsDownloadPage/index.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Redirect, useParams } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { logError } from '@edx/frontend-platform/logging';
+import { camelCaseObject } from '@edx/frontend-platform';
+import { ToastsContext } from '../Toasts';
+import LicenseManagerApiService from '../../data/services/LicenseManagerAPIService';
+
+const BulkEnrollmentResultsDownloadPage = ({ enterpriseId }) => {
+  const { enterpriseSlug, bulkEnrollmentJobId } = useParams();
+  const { addToast } = useContext(ToastsContext);
+  const [isLoading, setLoading] = useState(true);
+  const [redirectUrl, setRedirectUrl] = useState(null);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    if (enterpriseId && isLoading) {
+      LicenseManagerApiService.fetchBulkEnrollmentJob(enterpriseId, bulkEnrollmentJobId)
+        .then((response) => {
+          const result = camelCaseObject(response.data);
+          if (result.downloadUrl) {
+            setRedirectUrl(result.downloadUrl);
+          } else {
+            throw new Error('Your download is not ready yet.');
```

Comment:
[clarification/suggestion] How will users navigate to this page? I'm wondering if we could poll the API every so often  until the download is ready, with some messaging along those lines. Not sure if that'd be a good approach without knowing the full context of the page but seems like it might work well?

We do something similar in the `UserActivationPage` component, where we hydrate the authenticated user on an interval (via a handy `useInterval` hook) until a field in the response data comes back as true ([code example](https://github.com/edx/frontend-app-admin-portal/blob/master/src/components/UserActivationPage/index.jsx#L44-L48)). While it's not true, appropriate messaging is shown to inform user to wait and their download will start soon, for example.

Might not be the use case you're looking for here, but polling could be an option!

Reply:
They'll get the URL from a notification email once their download is ready. In theory, a user should never see this page before their download is ready.